### PR TITLE
Generate bitcode if HL_GENBITCODE environment variable is set.

### PR DESCRIPTION
**Pull request by alinas**

### CODE REVIEW

File: src/Util.cpp [hunk 1]
```diff
@@ -13,6 +18,45 @@ using std::vector;
 using std::ostringstream;
 using std::map;
 
+std::string get_env_variable(char const *env_var_name, size_t &read) {
+    if (!env_var_name)
```
Comment (zvookin) [inline src/Util.cpp]: Please always use braces.

File: src/Util.cpp [hunk 2]
```diff
@@ -13,6 +18,45 @@ using std::vector;
...
+    getenv_s(&read, lvl, env_var_name);
+    #else
+    char *lvl = getenv(env_var_name);
+    read = (lvl)?1:0;
+    #endif
+
+    if (read) {
+        return std::string(lvl);
+    }
+    else
```
Comment (zvookin) [inline src/Util.cpp]: Braces.

File: src/Util.cpp [hunk 3]
```diff
@@ -13,6 +18,45 @@ using std::vector;
...
+    #else
+    string program_name;
+    char path[PATH_MAX];
+    ssize_t len = ::readlink("/proc/self/exe", path, sizeof(path)-1);
+    if (len != -1) {
+        path[len] = '\0';
+        string tmp = std::string(path);
+        program_name = tmp.substr(tmp.find_last_of("/")+1);
+    }
+    else
```
Comment (zvookin) [inline src/Util.cpp]: Braces.

File: test/correctness/specialize.cpp [hunk 4]
```diff
@@ -394,7 +394,10 @@ int main(int argc, char **argv) {
         out.compute_root().specialize(cond1 && cond2).vectorize(x, 4);
 
         if_then_else_count = 0;
-        out.add_custom_lowering_pass(new CountIfThenElse());
+        CountIfThenElse *pass1 = new CountIfThenElse();
```
Comment (dsharletg) [inline test/correctness/specialize.cpp]: Now that this isn't being handed off to a custom lowering pass, it should probably just be a stack variable:

    CountIfThenElse pass1;
    pass1.mutate(...);
    ...

File: test/correctness/specialize.cpp [hunk 5]
```diff
@@ -420,7 +423,10 @@ int main(int argc, char **argv) {
         out.compute_root().specialize(cond1 && cond2).vectorize(x, 4);
 
         if_then_else_count = 0;
-        out.add_custom_lowering_pass(new CountIfThenElse());
+        CountIfThenElse *pass2 = new CountIfThenElse();
```
Comment (dsharletg) [inline test/correctness/specialize.cpp]: Same as above.

**Pull request merged**